rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,430 @@
+Any new message transmission at this stage (especially the locally published messages) gets delayed. 
+
+Adaptive message-forwarding can help such peers prioritize traffic to minimize latency for essential message transfers. 
+For instance, any GossipSub peer will likely receive every message from multiple senders, 
+leading to redundant transmissions [[11](https://ethresear.ch/t/number-duplicate-messages-in-ethereums-gossipsub-network/19921)]. 
+
+Implementing efficient strategies (only for slow senders) like lazy sending 
+and prioritizing locally published messages/IWANT replies over already queued messages 
+can help minimize outgoing message queue sizes and optimize bandwidth for essential message transfers. 
+
+A peer can identify itself as a slow peer by using any bandwidth estimation approach, 
+or simply setting an outgoing message queue threshold for all mesh members. 
+
+Eliminating/deprioritizing some messages can lower a peer's score, 
+but it also earns the peer an overall better score by achieving some early message transfers.  
+For instance, sending many near-first messages can only save a peer from a deficit penalty. 
+On the other hand, sending only one message (assuming MeshMessageDeliveriesThreshold defaults to 1) 
+as the first delivered message can add to the accumulative peer score. 
+
+### 2. Mitigating Transport Issues
+Congestion avoidance algorithms used in various TCP versions directly influence achievable throughput and message transfer time, 
+as maximum unacknowledged in-flight bytes are based on the congestion window $(C_{wnd})$ size. 
+
+Rapid adaptation of $C_{wnd}$ to the available network conditions can help lower message dissemination latency. 
+
+Therefore, selecting a more suitable TCP variant like BBR, 
+which is known for its ability to dynamically adjust the congestion window based on network conditions, 
+can significantly enhance GossipSub's performance. 
+
+At the same time, parameters like receive window scaling and initial $C_{wnd}$ also impact message transfer time, 
+but these are usually OS-specific system-wide choices.
+
+One possible solution is to raise $C_{wnd}$ by exchanging data over the newly established connection.
+This data may involve useful details like peer exchange information and gossip to build initial trust, 
+or GossipSub can use some dummy data to raise $C_{wnd}$ to a reasonable level.
+
+It's important to understand that some TCP variants reset $C_{wnd}$ after specific periods of inactivity [[12](https://datatracker.ietf.org/doc/html/rfc2581#section-4.1)].
+It can lead to a decline in TCP's performance for applications 
+that generate traffic after long enough intervals to trigger the resetting of the congestion window. 
+
+To address this, implementing straightforward measures like transport-level ping-pong messages can effectively mitigate this problem [[13](https://github.com/libp2p/specs/pull/558)]. 

```

Comment:
Remove "To address this" and start the sentence "Implementing straightforward measures...."